web/eula.go

```diff
@@ -0,0 +1,21 @@
+package web
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/trento/web/services"
+)
+
+func EulaAcceptHandler(settings services.SettingsService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		err := settings.AcceptEula()
+		if err != nil {
+			log.Error(err)
+			c.HTML(http.StatusInternalServerError, "error.html.tmpl", gin.H{"Error": "There was an error accepting the EULA. Please try again."})
+			return
+		}
+		c.Redirect(http.StatusFound, "/")
```

Comment:
Maybe we could put toast message or an flash alert banner on the top of the home page once the Eula is accepted to give more feedback.

Reply:
I tried to do that but actually I'm afraid both things are not viable:

- the flash alert won't be kept after the redirect, because the redirect makes the browser start a new request flow
- the toast message wouldn't live after the refresh, and we should do the whole thing with javascript to do that. I wanted to avoid javascript for such a simple feature.

Reply:
@dottorblaster  I can help  you with that I am sure we can do something about it, e.g. save the flash in session

Reply:
We already have the flash message mechanism in place in `web/alert.go`

Reply:
If you folks meant `StoreAlert` I tried and nothing happened :smile: